codeflare/pipelines/Runtime.py

```diff
@@ -107,8 +107,8 @@ def execute_or_node_remote(node: dm.EstimatorNode, mode: ExecutionType, xy_ref:
     elif mode == ExecutionType.SCORE:
         if base.is_classifier(estimator) or base.is_regressor(estimator):
             estimator = node.get_estimator()
-            res_Xref = ray.put(estimator.score(X, y))
-            result = dm.XYRef(res_Xref, xy_ref.get_yref(), prev_node_ptr, prev_node_ptr, [xy_ref])
+            score_ref = ray.put(estimator.score(X, y))
+            result = dm.XYRef(score_ref, score_ref, prev_node_ptr, prev_node_ptr, [xy_ref])
```

Comment:
so, this should just be fine when going downstream as a `transform` results in Y and is sent downstream, correct @yuanchi2807 ?

Reply:
It is executed under ExecutionType.SCORE. Could `transform` happen downstream? @raghukiran1224

Reply:
I had a similar question as @raghukiran1224 asked.  @yuanchi2807 Even though it is executed under `ExecutionType.SCORE,` but a node can still be in the middle of a pipeline, I think. Hence, it can still be sent downstream.

Reply:
Good question raised. If we follow sklearn convention, predictor.predict(data) and model.score(data) return results semantically different from pipeline input. We could change data interpretation in the middle of a pipeline but we'd be introducing new semantics. There is no right or wrong but a design decision to conform to sklearn convention. @klwuibm 
https://scikit-learn.org/stable/developers/develop.html

Reply:
so, we have to be careful here -- sklearn convention outside is one thing -- the output mapping, which could be done through modifications to `XYRef`, and this is changing pipeline runtime semantics.

@mudhakar Thoughts?

Reply:
Should the question being asked be: Can we tell if a node is a terminal node at the end of the pipeline?  If yes, then we make the changes that @yuanchi2807 did.  Otherwise, we don't, because a node can be in the middle?

Reply:
We may consider in the context of its usage. 
`execute_pipeline(fitted_pipeline, ExecutionType.SCORE/ExecutionType.PREDICT, test_input)`
so when a score/predictions are fetched, semantically should it come from Xref or yref?

Reply:
@klwuibm @raghukiran1224 
I updated execute_or_node_remote() to only set score or predict results at an output node where applicable, per discussion. Thanks.

Reply:
lgtm;